protos/bottle/assembly/v1/build.proto

```diff
@@ -24,6 +24,7 @@ message Build {
   }
 
   repeated BuildComponent build_components = 3;
+  repeated BuildComponent missing_components = 8;
```

Comment:
There's pros and cons to this approach so I'm open to ideas.

A pro is that we are able to continue dealing with just builds between services rather than builds sometimes, then components other times, as returned types. A con is that this likely won't be populated by many services so many, it will just be empty.

Not including it here will require other changes as a result:

1. Adding a new gRPC command to Bottle for missing components
2. Adding the aforementioned command to Assembly
3. Updating the UI to make a second call for this information for each build it needs it for

Reply:
This feels more similar to what we are currently doing